dds/DCPS/GuidUtils.h

```diff
@@ -274,6 +272,28 @@ GUID_t bit_key_to_guid(const DDS::BuiltinTopicKey_t& key)
   return id;
 }
 
+struct OpenDDS_Dcps_Export GuidPair {
+  GUID_t local;
+  GUID_t remote;
+
+  GuidPair(const GUID_t& local, const GUID_t& remote)
+  : local(local)
+  , remote(remote)
+  {
+  }
+
+  int cmp(const GuidPair& other) const
```

Comment:
I think my biggest concern here is that we'll eventually want to optimize comparisons on some of these, and so a packed data structure makes sense (so we can use a single call to `std::memcmp`). But maybe that just needs to be a separate data structure if we ever get back to that point.

Reply:
Does inline end up producing the same code?  godbolt.org may be helpful with this.

Reply:
> Does inline end up producing the same code? godbolt.org may be helpful with this.

- Single `memcmp`: https://godbolt.org/z/Mq35aMW7K
- 1 or 2 `memcmp`: https://godbolt.org/z/Y31EeqGrn

At `-O1` and higher, they look like they inline and optimize about as much as they can, but they don't take the leap that Tim wants. GCC and MSVC look the basically the same, but MSVC still includes the unused template instance definitions for some reason, even at higher `-O` levels. I haven't gone through Clang line by line to completely understand what it's doing, but it's interesting because it looks like it's inlining an unrolled implementation of `memcmp`.

If we can get away with this for now I'd prefer it so we're not prematurely optimizing. If it turns out this is a problem, we can change it later or bring back something like the packed `RoutedGuidPair`. As it is this PR should increase performance because, excluding the broken shmem `GuidPair` comparison, the other two `GuidPair`s were using less than calls.

Reply:
OK, let's leave it like this for now but make a mental note to revisit if we see these being used as map keys in performance-critical maps, etc.